### PR TITLE
DOC: add a FAQ for cluster startup issues

### PR DESCRIPTION
**Pull request by mattip**
## Why are these changes needed?

Part of issue #34094. This adds a FAQ to the VM UserGuide.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by mattip** [comment]
The [rendered user guide doc](https://anyscale-ray--36015.com.readthedocs.build/en/36015/cluster/vms/user-guides/index.html#vm-cluster-guides) now has a [new FAQ](https://anyscale-ray--36015.com.readthedocs.build/en/36015/cluster/vms/user-guides/faq.html) entry.

File: doc/source/cluster/vms/user-guides/faq.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. _cluster-FAQ:
+
+FAQ
+===
+
+What is the difference between ``--node-ip-address`` and ``--address``?
+-----------------------------------------------------------------------
+
+When starting a head node, ``ray start --node-ip-address xx.xx.xx.xx --port
```
Comment (architkulkarni) [inline doc/source/cluster/vms/user-guides/faq.rst:9]:
```suggestion
When starting a head node, ``ray start --head --node-ip-address xx.xx.xx.xx --port
```
Should we add `--head` to this command?

Also, a user might wonder when it is necessary to specify `--node-ip-address` and what happens when it isn't specified.  The naive question is why do we need to tell the head node its own ip address, and I guess the answer is that it's an *externally reachable* IP address, and Ray needs to know it to allow worker nodes to connect?

File: doc/source/cluster/vms/user-guides/faq.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+* Using a machine with a large number of cpus, and starting one worker per CPU
+  together with OpenBLAS (as used in NumPy) may allocate too many threads. This
+  is an `known OpenBLAS limitation`_ and can be mitigated by limiting OpenBLAS
+  to one thread per process as explained in the link.
+
+* Resource allocation is not what was expected: usually too many CPUs per node
+  were allocated. Best practice is to verify your SLURM configuration without
+  starting ray to verify that the allocations are as expected. For more
+  detailed information see :ref:`ray-slurm-deploy`
```
Comment (architkulkarni) [inline doc/source/cluster/vms/user-guides/faq.rst:54]:
```suggestion
  detailed information see :ref:`ray-slurm-deploy`.
```

File: doc/source/cluster/vms/user-guides/faq.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+I am having problems getting my SLURM cluster to work
+-----------------------------------------------------
+
+There seem to be a class of issues starting ray on SLURM clusters.  While we
+have not been able to pin down the exact causes (as of June 2023), work has
+been done to mitigate some of the resource contention. Some of the issues
+reported:
+
+* Using a machine with a large number of cpus, and starting one worker per CPU
```
Comment (architkulkarni) [inline doc/source/cluster/vms/user-guides/faq.rst:46]:
```suggestion
* Using a machine with a large number of CPUs, and starting one worker per CPU
```

File: doc/source/cluster/vms/user-guides/faq.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+When starting a head node, ``ray start --node-ip-address xx.xx.xx.xx --port
+nnnn`` is used to specify the address of the head node. When starting a worker
+node, ``ray start --address xx.xx.xx.xx:nnnn`` is used to specify the head node
+this worker should connect to.
+
+What does a worker node failure to connect look like?
+-----------------------------------------------------
+
+If the worker node cannot connect to the head node, you should see this error:
+::
```
Comment (architkulkarni) [inline doc/source/cluster/vms/user-guides/faq.rst:18]: Screenshot 2023-06-02 at 4 58 10 PM
It's using code syntax highlighting by default, can we specify "text" somehow?

**Review by architkulkarni — COMMENTED** [review]
Looks good overall, just a few nits. Thanks for adding this!

File: doc/source/cluster/vms/user-guides/faq.rst (hunk 1 above)
Comment (mattip) [reply doc/source/cluster/vms/user-guides/faq.rst]: I reworked this section and related to the other review comments.

File: doc/source/cluster/vms/user-guides/faq.rst (hunk 4 above)
Comment (mattip) [reply doc/source/cluster/vms/user-guides/faq.rst]: The best I could do was "quote"

**Comment by architkulkarni** [comment]
Looks good to me, doctest failure ` >>> ds = ray.data.range_tensor(1, shape=(64, 64))` unrelated

**Review by architkulkarni — APPROVED** [review]

**Comment by architkulkarni** [comment]
Oh sorry, I just noticed we have another clusters FAQ here: https://anyscale-ray--36015.com.readthedocs.build/en/36015/cluster/faq.html
Do you think it's better to merge these, or keep them separate?


If we keep them separate do you mind adding it to the sidebar below? It seems to be missing currently. Let me know if you need help finding where to add this (I don't remember off the top of my head unfortunately)

Screenshot 2023-06-05 at 9 22 28 AM

**Comment by jjyao** [comment]
> Do you think it's better to merge these, or keep them separate?

I think it's better to merge these: at least some of the FAQs apply to KubeRay as well

**Comment by mattip** [comment]
I have merged the two FAQs to `cluster/faq.html` and referenced the single document from `cluster/vms/user-guides/index.html` as well.

**Comment by architkulkarni** [comment]
Screenshot 2023-06-07 at 9 48 26 AM
Thanks, looks good!

**Pull request merged**